src/chapters/more-on-functions/receiving-functions.rst

```diff
@@ -0,0 +1,210 @@
+      Enter an even number: 5
+      Invalid input. Try again.
+      Enter an even number: 4
+      4
+
+The function ``getValidInput`` handles the work of interacting with the user, while allowing the validation logic to be customized. This separates the different concerns of validation and user interaction, sticking to the idea that *a function should do only one thing*. It also enables more reusable code. If we need to get different input from the user, we can simply call ``getValidInput`` with different arguments.
+
+.. admonition:: Example
+
+   This example uses the same ``getValidInput`` function defined above with a different prompt and validator function. In this case, we check that a potential password has at least 8 characters.
+
+   .. sourcecode:: js
+      :linenos:
+
+      const input = require('readline-sync');
+
+      function getValidInput(prompt, isValid) {
+         
+         let userInput = input.question(prompt);
+
+         while (!isValid(userInput)) {
+            console.log("Invalid input. Try again.");
+            userInput = input.question(prompt);
+         }
+
+         return userInput;
+      }
+
+      let isValidPassword = function(password) {
+
+         // Passwords should have at least 8 characters
+         if (password.length < 8) {
+            return false;
+         }
+
+         return true;
+      };
+
+      console.log(getValidInput('Create a password:', isValidPassword));
+
+   **Sample Output**
+
+   ::
+
+      Create a password: launch
+      Invalid input. Try again.
+      Create a password: code
+      Invalid input. Try again.
+      Create a password: launchcode
+      launchcode
+
+.. admonition:: Try It!
+
+   #. Use our ``getValidInput`` function to ensure user input starts with "a".
+   #. Create another validator that ensures user input is a vowel.
+
+   `Try it at repl.it <https://repl.it/@launchcode/Validator-check>`_
+
+
+Example: A Logger
+-----------------
+
+Another common example of a function using another function to customize its behavior is that of logging. Real-world applications are capable of logging messages such as errors, warnings, and statuses. Such applications allow for log messages to be sent to one or more destinations. For example, the application may log messages to both the console and to a file.
+
+We can write a logging function that relies on a function parameter to determine the logging destination.
+
+A Simple Logger
+^^^^^^^^^^^^^^^
+
+.. admonition:: Example
+
+   The ``logError`` function outputs a standardized error message to a location determined by the parameter ``logger``.
+
+   .. sourcecode:: js
+      :linenos:
+   
+      let fileLogger = function(msg) {
+
+         // Put the message in a file
+
+      }
+
+      function logError(msg, logger) {
+         let errorMsg = 'ERROR: ' + msg;
+         logger(errorMsg);
+      }
+
+      logError('Something broke!', fileLogger);
+
+Let's examine this example in more detail.
```

Comment:
This is much better than before. Your description of how the code works is clear and easy to follow, and it keeps these examples from being too heavy of a lift for the students.